packages/combinators/trycatch.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Runs a boolean check on the input arguments (an action),
+// if true (any non-error value), runs an action, else, returns {}.
+// Example invocation:
+//   $ wsk action invoke -br eca \
+//          -p '$conditionName' check \
+//          -p '$actionName' test \
+//          -p x 12 -p password 'ok'
+
+var openwhisk = require('openwhisk')
+
+function main (args) {
+  const wsk = openwhisk({ignore_certs: args.ignore_certs || true})
+
+  const tryName = args['$tryName']
+  const catchName = args['$catchName']
+
+  delete args['$tryName']
+  delete args['$catchName']
+
+  if (typeof tryName !== 'string') {
+    return { 'error': "Expected an argument '$tryName' of type 'string'." }
+  }
+
+  if (typeof catchName !== 'string') {
+    return { 'error': "Expected an argument '$catchName' of type 'string'." }
+  }
+
+  return wsk.actions
+  .invoke({
+          actionName: tryName,
+          params: args,
+          blocking: true
+      })
+  .then(activation => {return activation.response.result})
```

Comment:
minor style point. this could be `.then(activation => activation.response.result)`